### ISSUE TITLE
Render after updating smart object

### ISSUE THREAD

**Issue by YCanvas** [Agamnentzar/ag-psd#138]
@Agamnentzar Thank you for this package. It is awesome.

Upon reading through documentations and looking through issues, i see that the package doesn't support updating the bitmap of the prerendered image after updating the smart objects. However, i see that you kept advising users a way around it by using node-canvas to update the bitmap then it is possible to export new rendered image.

Would you please provide a small example of this? like a quick simple small code with a psd example file?

I appreciate it and thank you

**Comment by Agamnentzar** [comment]
You can see canvas tutorials here on MDN: https://developer.mozilla.org/en-US/docs/Web/API/Canvas_API/Tutorial

The only difference with node-canvas is that instead of creating `<canvas>` element you create new canvas with `createCanvas` function provided by node-canvas, you can read more here: https://github.com/Automattic/node-canvas

after you create new canvas you just replace `layer.canvas` property and adjust `layer.left` and `layer.top` if needed.

**Comment by YCanvas** [comment]
@Agamnentzar Thank you so much for the quick response and support.

I tried this code provided by [zkschmitz] here https://github.com/Agamnentzar/ag-psd/issues/128
I think the code follows your recommendations, but still outputs the pre-rendered image, do you see any issues with the code he provided?

**Comment by Agamnentzar** [comment]
by "outputs" you mean the image that you see in Photoshop ? The code provided looks good, it replaces layer.canvas to a different image.

**Comment by YCanvas** [comment]
sorry for the confusion, i meant that when i run the code, the "altered-image.png" image exported is not the updated one, but the pre-rendered image (The image i see first time i open the psd on photoshop)

**Comment by Agamnentzar** [comment]
Yes, the library doesn't re-render the composite image, you have to re-render it yourself. It's not required for the PSD document to work in Photoshop but you need to redraw it, by compositing all the layers with correct blend modes and applying all required effects if you want to be able to use it.

It seems what you want is to generate PNG image and not PSD document, in that case your approach might not be the best solution, it might be better to just draw the PNG you need using node-canvas.

If you want to modify PSD file and then export PNG composite image then this library will not help you much.

**Comment by jaankoppe** [comment]
> sorry for the confusion, i meant that when i run the code, the "altered-image.png" image exported is not the updated one, but the pre-rendered image (The image i see first time i open the psd on photoshop)

Hey @YCanvas, maybe you can use Mediamodifier Mockup API for this?
This will lead you to a similar issue another user had: https://github.com/Agamnentzar/ag-psd/issues/147#issuecomment-1663964082
